### PR TITLE
Cleanup batch_normalization

### PR DESCRIPTION
**Pull request by sonots**
This is trivial cleanup of batch_normalization codes. This does not change any behaviors.

1. Rename `param_ndim` to `gamma_ndim` to make meaning be more explicit.
2. Add a comment to explain what is `key_axis`.
3. Cleanup `_as4darray` logic to make meaning be more explicit.
4. Remove `_get_mode` which is not used from anywhere.

### CODE REVIEW

File: chainer/functions/normalization/batch_normalization.py [hunk 1]
```diff
@@ -658,20 +659,13 @@ def can_use_cudnn(self, xp):
...
-def _as4darray(arr, key_axis):
-    if arr.ndim == 4 and key_axis[0] == 1:
+def _as4darray(arr, mode):
+    assert mode.cudnn_dim_ok
+    if mode.is_for_conv2d:
+        assert arr.ndim == 4
         return arr
-    elif key_axis[0] == arr.ndim - 1:
+    else:  # is_for_linear
         return arr.reshape(numpy.prod(arr.shape[0:-1]), -1, 1, 1)
```
[the remark this answers is not in the document]
Comment (sonots) [reply chainer/functions/normalization/batch_normalization.py]: In my understanding it is different.

```
import numpy

a = np.arange(24).reshape(2, 3, 4)
In [5]: len(a)
Out[5]: 2
In [7]: a.shape[0:-1]
Out[7]: (2, 3)
In [9]: np.prod(a.shape[0:-1])
Out[9]: 6
```

[the remark this answers is not in the document]
Comment (okuta) [reply chainer/functions/normalization/batch_normalization.py]: OK, This is my miss.

**Comment by sonots** [comment]
chainer-doc on `examples/cnn.rst` failed with ssl handshake, but not related with this PR.

**Comment by sonots** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
`batch_normalization.py` was fixed by many PRs.

**Pull request merged**